### ISSUE TITLE
TraitsUI 7.3.1 Release

### ISSUE THREAD

**Issue by corranwebster** [enthought/traitsui#1870]
Target date: 

Release by 

Release blockers
----------------
None

Pre-release
---
- [x] Verify that no other open issue needs to be addressed before the release.
- [x] Test against other ETS packages and other ETS-using projects
- [x] Check MANIFEST, requirements, changelog are still up to date.
- [x] Test building the documentation

Release
-------
- [x] Create branch `release/<release version number>` from `maint/7.3` branch.
  - [x] Set release to `<release version number>`, and set `IS_RELEASED` is true; commit
  - [x] Install from source distribution and run tests again
  - [x] Open a PR against `maint/7.3` with this being the last commit so that CI is built on the release commit
  - [x] Once CI is done building merge PR
- [x] Branch and bump the minor version number i.e. `<release version number + 0.1>` and set `IS_RELEASED` to false; commit.
  - [x] Open and then merge a separate PR against `maint/7.3`
- [x] From the commit at which `IS_RELEASED` is true and version is `<release version number>`, tag (annotated!) `git tag -a -m "Release <release version number>" <release version number>`
  - [x] Push the tag `git push origin <release version number>`
- [x] Make PR targeting `gh-pages` branch: Generate documentation and copy the content to the branch. Verify that the resulting index.html looks good.
- [x] Upload to PyPI
- [x] Test the PyPI package

Post-release
-------------
- [x] Package update for `enthought/free` repository (for EDM)
- [x] Backport release note and change log to ``main``, and possibly `maint/<release version number>` branch.
- [x] Update GitHub Release pages https://github.com/enthought/traitsui/releases
- [ ] Announcement (e.g. Google Group)
